### ISSUE TITLE
`mc ls` dir give strange result when the dir path ended with `/`

### ISSUE THREAD

**Issue by iwanbk** [gigforks/minio#24]
```bash
$ mc ls myzerostor/keruh/oss/
[2018-03-08 08:56:33 WIB]  30KiB /keruh/gateway-oss.go
[2018-03-08 08:56:33 WIB] 5.4KiB /keruh/gateway-oss_test.go
ibk@siputih:~/pathgo/src/github.com/minio/minio/cmd/gateway/zerostor
$ mc ls myzerostor/keruh/oss
[2018-03-08 08:56:33 WIB]  30KiB gateway-oss.go
[2018-03-08 08:56:33 WIB] 5.4KiB gateway-oss_test.go
```

**Comment by iwanbk** [comment]
There is no problem in web UI

**Comment by iwanbk** [comment]
From further checks, minio call `ListObjects` three times when try to list a directory.

First and second: without trailing slash.
Third: without trailing slash.

backblaze gateway only responded to the last call.
While current zerostor gateway respon to all

**Comment by iwanbk** [comment]
fixed by e101e4a9bc89153dbe0879439e01b6a1c1dd0473
